### ISSUE TITLE
add a row for which cms support content blocks/slices

### ISSUE THREAD

**Issue by kelenakamura** [gentics/headless-cms-comparison#9]
so the ability to build a page based on blocks of fields (like prismics content slices - https://prismic.io/feature/dynamic-layout-content-components

**Comment by Jotschi** [comment]
I'm not sure whether this would be a quantifiable feature. Adding nested content (I assume this would be similar to a content slice) can easily be done by creating references between content objects. There seems to be multiple ways to solve this requirement (reuse of existing content elements).

Does prismic provide a dedicated API for this?

**Comment by kelenakamura** [comment]
we just see a lot of clients like the idea of dynamically building pages out of content blocks/slices.

basically a way to build a page via a list of modules/components that you can select to add to a page.

so rather than a defined content type like blog post with a set of specific fields (title, image, body etc). you could build a blog post using (a video component, a carousel, a text block, a quote etc). in whatever order you would like.

so just a way to build dynamic page content based on pre-defined content blocks/slices/modules/components (or whatever they need to be called).

so if this is a requirement for a client was hoping there was a way to see which headless CMS’s could support that. currently we only know that prismic supports it.

**Comment by Jotschi** [comment]
In Gentics Mesh it is possible to create microschemas. A microschema is a mini template which contains a subset of fields. This allows you to add extra elements to your contents. A content like a blog can have a micronode list. That list can contain multiple micronodes which make use of the microschemas. That way it is possible to add (e.g. a vcard, video element, geolocation).

vcard:
* name
* surname
* adress.

geolocation:
* lon
* lat 

video:
* title
* thumbnail
* video binary (contains filename, length, resolution)


Would that be a comparable feature?

**Comment by kelenakamura** [comment]
yes it sounds like a similar feature. basically as long as you can have a single content type (ie blog post) but with the use of these self contained content blocks can create numerous customized pages depending on which blocks you decide to add to the page during content entry.

**Comment by Stupidism** [comment]
I think the key is to embed content blocks inside RichText. So it looks like an article in the editor instead of several content blocks located among more broken paragraphs.

Contentful has this feature. And I didn't see any other CMS has this feature until I read this issue.

**Comment by Stupidism** [comment]
BTW, it looks like this: 
<img width="819" alt="image" src="https://user-images.githubusercontent.com/5801015/160759540-d9f43573-53c0-450e-b08a-40ba47bc859d.png">
